xdsl/ir/core.py

```diff
@@ -542,16 +542,22 @@ def is_valid_name(cls, name: str | None):
         return name is None or cls._name_regex.fullmatch(name)
 
     @staticmethod
-    def get(arg: SSAValue | Operation) -> SSAValue:
+    def get(
+        arg: SSAValue | Operation, *, type: type[AttributeInvT] = Attribute
```

Comment:
is there any way to have a `TypeForm` here instead of a `type`? There was a case where it would be good for the type to be `MemRefType | UnrankedMemRefType` but the `isinstance` check below wasn't happy

Reply:
Why not, have you tried it? I'd actually do an isa then instead of isinstance

Reply:
I get 
```
Argument of type "TypeForm[AttributeInvT@get]" cannot be assigned to parameter "class_or_tuple" of type "_ClassInfo" in function "isinstance"
```
I could try `isa`

Reply:
It turns out the `init` for `ToTensorOp` (which is what I wanted this for, doesn't actually make any sense as it checks that an `UnrankedMemRefType` is a `ShapedType`

Reply:
Worth a fix in a separate PR IMO

Reply:
> no way to get a name out for the error message
What do you mean by this? Having an `isa` would be really cool here!

Reply:
I currently put `type.name` in the error message but this doesn't work if `type` is a `TypeForm`, so the cost of allowing a `TypeForm` would be to make the error message worse I think (unless you can see a way round this.

Reply:
#4431 for init fix